src/v/cloud_storage_clients/s3_client.cc

```diff
@@ -995,4 +992,85 @@ ss::future<> s3_client::do_delete_object(
       });
 }
 
+static auto iobuf_to_delete_objects_result(iobuf&& buf) {
+    auto root = iobuf_to_ptree(std::move(buf));
+    auto result = client::delete_objects_result{};
+    try {
+        for (auto const& [tag, value] : root.get_child("DeleteResult")) {
+            if (tag != "Error") {
+                continue;
+            }
+            auto code = value.get_optional<ss::sstring>("Code");
+            auto key = value.get_optional<ss::sstring>("Key");
+            auto message = value.get_optional<ss::sstring>("Message");
+            auto version_id = value.get_optional<ss::sstring>("VersionId");
+            vlog(
+              s3_log.trace,
+              R"(delete_objects_result::undeleted_keys Key:"{}" Code: "{}" Message:"{}" VersionId:"{}")",
+              key.value_or("[no key present]"),
+              code.value_or("[no error code present]"),
+              message.value_or("[no error message present]"),
+              version_id.value_or("[no version id present]"));
+            if (key.has_value()) {
+                result.undeleted_keys.push_back({
+                  object_key{key.value()},
+                  code.value_or("[no error code present]"),
+                });
+            } else {
+                vlog(
+                  s3_log.warn,
+                  "an DeleteResult.Error does not contain the Key tag");
+            }
+        }
+    } catch (...) {
+        vlog(
+          s3_log.error,
+          "DeleteObjects response parse failed: {}",
+          std::current_exception());
+        throw;
+    }
+    return result;
+}
+
+auto s3_client::do_delete_objects(
+  bucket_name const& bucket,
+  std::span<const object_key> keys,
+  ss::lowres_clock::duration timeout)
+  -> ss::future<client::delete_objects_result> {
+    auto request = _requestor.make_delete_objects_request(bucket, keys);
+    if (!request) {
+        return ss::make_exception_future<delete_objects_result>(
+          std::system_error(request.error()));
+    }
+    auto& [header, body] = request.value();
+    vlog(s3_log.trace, "send DeleteObjects request:\n{}", header);
+
+    return ss::do_with(
+             std::move(body),
+             [&_client = _client, header = std::move(header), timeout](
```

Comment:
i'm not against this pattern `&_client = _client` for any technical reason, but it is more idiomatic in our code base to capture `this` which IMO makes it more visually obvious the scope / lifetime of what is being captured.